### ISSUE TITLE
classInt seems to cause edge case error in sf plot

### ISSUE THREAD

**Issue by Robinlovelace** [r-spatial/classInt#16]
Was going to report in the `sf` issue track but on reflection I think the fix is more likely to be found here. Reprex below:

``` r
library(sf)
#> Linking to GEOS 3.7.1, GDAL 2.4.2, PROJ 5.2.0
crashes_sf = readr::read_csv("https://github.com/ropensci/stats19/releases/download/1.0.0/crashes.csv")
#> Parsed with column specification:
#> cols(
#>   vehicle_type = col_character(),
#>   casualty_type = col_character(),
#>   casualty_age = col_double(),
#>   dark = col_logical(),
#>   casualty_day = col_date(format = ""),
#>   casualty_hms = col_character(),
#>   casualty_hour = col_double()
#> )
crashes_sf$longitude = c(-1.3, -1.2, -1.1)
crashes_sf$latitude = c(50.7, 50.7, 50.68)
crashes_sf = st_as_sf(crashes_sf, coords = c("longitude", "latitude"), crs = 4326)
plot(crashes_sf)
#> Error in classInt::classIntervals(v0, min(nbreaks, n.unq), breaks, warnSmallN = FALSE): var is not numeric
mapview::mapview(crashes_sf) # for interactive map 
```

![](https://i.imgur.com/ZJPKFSc.png)![](https://i.imgur.com/VCACYU0.png)

<sup>Created on 2019-10-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by Robinlovelace** [comment]
Update: it's the date column that's causing the Error.

**Comment by edzer** [comment]
I recall there is now support for that; did you try the versions from github (both sf and classInt)?

**Comment by Robinlovelace** [comment]
Aha, no only tried dev version of sf. That fixes it, thanks!

``` r
remotes::install_github("r-spatial/sf")
#> Skipping install of 'sf' from a github remote, the SHA1 (cd292c78) has not changed since last install.
#>   Use `force = TRUE` to force installation
remotes::install_github("r-spatial/classInt")
#> Skipping install of 'classInt' from a github remote, the SHA1 (f35b0d70) has not changed since last install.
#>   Use `force = TRUE` to force installation
library(sf)
#> Linking to GEOS 3.7.1, GDAL 2.4.2, PROJ 5.2.0
crashes_sf = readr::read_csv("https://github.com/ropensci/stats19/releases/download/1.0.0/crashes.csv")
#> Parsed with column specification:
#> cols(
#>   vehicle_type = col_character(),
#>   casualty_type = col_character(),
#>   casualty_age = col_double(),
#>   dark = col_logical(),
#>   casualty_day = col_date(format = ""),
#>   casualty_hms = col_character(),
#>   casualty_hour = col_double()
#> )
crashes_sf$longitude = c(-1.3, -1.2, -1.1)
crashes_sf$latitude = c(50.7, 50.7, 50.68)
crashes_sf = st_as_sf(crashes_sf, coords = c("longitude", "latitude"), crs = 4326)
plot(crashes_sf)
```

![](https://i.imgur.com/iiFQnE0.png)

``` r
mapview::mapview(crashes_sf) # for interactive map 
```

![](https://i.imgur.com/la7Z8DJ.png)

<sup>Created on 2019-10-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by rsbivand** [comment]
Will release shortly, on resolution of a broken URL and https://github.com/r-spatial/classInt/issues/15.

**Comment by rsbivand** [comment]
0.4-2 submitted to CRAN.

**Comment by Robinlovelace** [comment]
Thanks for the fix + update @rsbivand may be useful for the [R for Road Safety](https://www.racfoundation.org/introduction-to-r-for-road-safety) course I'm teaching!

**Comment by rsbivand** [comment]
Published 0.4-2
